libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2195,14 +2195,28 @@ void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCa
 	)
 	{
 		string msg = "Expected regular external function type, or external view on public function.";
-		if (externalFunctionType->kind() == FunctionType::Kind::Internal)
-			msg += " Provided internal function.";
-		else if (externalFunctionType->kind() == FunctionType::Kind::DelegateCall)
-			msg += " Cannot use library functions for abi.encodeCall.";
-		else if (externalFunctionType->kind() == FunctionType::Kind::Creation)
-			msg += " Provided creation function.";
-		else
-			msg += " Cannot use special function.";
+
+		switch (externalFunctionType->kind())
+		{
+			case FunctionType::Kind::Internal:
+				msg += " Provided internal function.";
+				break;
+			case FunctionType::Kind::DelegateCall:
+				msg += " Cannot use library functions for abi.encodeCall.";
+				break;
+			case FunctionType::Kind::Creation:
+				msg += " Provided creation function.";
+				break;
+			case FunctionType::Kind::Event:
+				msg += " Cannot use event function.";
+				break;
+			case FunctionType::Kind::Error:
+				msg += " Cannot use error function.";
```

Comment:
Still needs a small tweak.
```suggestion
			case FunctionType::Kind::Event:
				msg += " Cannot use events for abi.encodeCall.";
				break;
			case FunctionType::Kind::Error:
				msg += " Cannot use errors for abi.encodeCall.";
```